src/main.js

```diff
@@ -1,26 +1,16 @@
 import { $ } from "./utils/dom";
 import header from "./components/header";
 import Modal from "./components/common/modal";
-import Title from "./components/common/title";
-import RegisterForm from "./components/registerForm";
-import renderRestaurants from "./domain/renderRestaurant";
-import { registerModalClose } from "./components/common/modal/handleCloseModal";
-import { restaurantList } from "./restaurantList";
+import CategoryAndSortFilter from "./components/categoryAndSortFilter";
+import FavoriteTabFilters from "./components/favoriteTabFilter";
+import Restaurants from "./model/Restaurants.ts";
 
 addEventListener("load", () => {
-  $("#app").prepend(header());
-  renderRestaurants(...restaurantList);
+  const restaurantList = new Restaurants();
+  $("#app").prepend(header(restaurantList.addRestaurant));
 
-  $("main").appendChild(
-    Modal({
-      handleClose: registerModalClose,
-      headerComponent: Title(
-        "새로운 음식점",
-        "h2",
-        "modal-title",
-        "text-title"
-      ),
-      bodyComponent: RegisterForm(),
-    })
-  );
+  $("main").prepend(CategoryAndSortFilter(restaurantList.changeState));
+  $("main").prepend(FavoriteTabFilters(restaurantList.changeState));
```

Comment:
```suggestion
  $("main").prepend(CategoryAndSortFilter({
    onSortByCategory: (category) => restaurantList.changeState({ category }),
    onSortByOption: (option) => restaurantList.changeState({ option }),
  }));
  $("main").prepend(FavoriteTabFilters({
    onSortByFavorate: (favorites)=> restaurantList.changeState({ favorites })
  }));
```
restaurantList.changeState 는 수행할 수 있는 일의 범주가 매우 넓기 때문에 이 메서드 자체를 매개변수로 넘겨주기 보다는 할 수 있는 일을 제한시켜서 넘겨주는 것이 좋을 것 같습니다. 또한, 넘긴 메서드가 정확히 무엇을 위해 사용될 지를 객체 키를 통해 나타내는 것이 제 생각에는 더 분명한 의도를 전달할 수 있을 것 같습니다.